spec/models/academic_year_spec.rb

```diff
@@ -12,18 +12,54 @@
 require "rails_helper"
 
 RSpec.describe AcademicYear, type: :model do
-  subject(:academic_year) do
-    build(:academic_year,
-          name: "2023 to 2024",
-          starts_on: Date.current,
-          ends_on: Date.current + 1.day)
-  end
+  describe "with validations" do
+    subject(:academic_year) do
+      build(:academic_year,
+            name: "2023 to 2024",
+            starts_on: Date.current,
+            ends_on: Date.current + 1.day)
+    end
 
-  context "with validations" do
     it { is_expected.to validate_presence_of(:name) }
     it { is_expected.to validate_presence_of(:starts_on) }
     it { is_expected.to validate_presence_of(:ends_on) }
-
     it { is_expected.to validate_comparison_of(:ends_on).is_greater_than_or_equal_to(:starts_on) }
   end
+
+  describe ".for_date" do
+    let!(:existing_academic_year) do
+      create(:academic_year,
+             name: "2024 to 2025",
+             starts_on: Date.new(2024, 9, 1),
+             ends_on: Date.new(2025, 8, 31))
```

Comment:
```suggestion
             starts_on: Date.parse("1 September 2024"),
             ends_on: Date.parse("1 August 2025"))
```
Using strings over dates improves readability